lib/spack/spack/cmd/uninstall.py

```diff
@@ -106,33 +106,41 @@ def find_matching_specs(env, specs, allow_multiple_matches=False, force=False):
     return specs_from_cli
 
 
-def installed_dependents(specs):
+def installed_dependents(specs, env):
     """Map each spec to a list of its installed dependents.
 
     Args:
         specs (list): list of Specs
+        env (Environment): the active environment, or None
 
     Returns:
-        (dict): mapping from spec to lists of Environments
+        (tuple): mapping from spec to sets of dependent specs
 
     """
-    dependents = {}
-    for item in specs:
+    local_dpts = {}
+    external_dpts = {}
+
+    env_hashes = set(env.all_hashes()) if env else set()
+
+    for spec in specs:
         installed = spack.store.db.installed_relatives(
-            item, 'parents', True)
-        lst = [x for x in installed if x not in specs]
-        if lst:
-            lst = list(set(lst))
-            dependents[item] = lst
-    return dependents
+            spec, direction='parents', transitive=True)
+
+        # separate installed dependents into dpts in this environment and
+        # dpts that are outside this environment
+        for dpt in installed:
+            if dpt not in specs:
+                if not env or dpt.dag_hash() in env_hashes:
+                    local_dpts.setdefault(spec, set()).add(dpt)
```

Comment:
I think `active_dpts` and `inactive_dpts` may be more consistent for naming (to avoid confusion with external packages)

Reply:
done